### DISCUSSION TITLE
MZKeychainWrapper.set(Int, ....) is will recurse forever

### DISCUSSION THREAD

**Discussion by gutley** [mozilla-mobile/firefox-ios#13734] [General]
Not sure who looks after MozillaRustComponents, but this func in MZKeychainWrapper needs some work:

```swift
    @discardableResult open func set(_ value: Int, forKey key: String, withAccessibility accessibility: MZKeychainItemAccessibility? = nil, isSynchronizable: Bool = false) -> Bool {
        return set(Int(truncating: NSNumber(value: value)), forKey: key, withAccessibility: accessibility, isSynchronizable: isSynchronizable)
    }
```
as it calls itself instead of presumably being this:
```swift
    @discardableResult open func set(_ value: Int, forKey key: String, withAccessibility accessibility: MZKeychainItemAccessibility? = nil, isSynchronizable: Bool = false) -> Bool {
        return set(NSNumber(value: value), forKey: key, withAccessibility: accessibility, isSynchronizable: isSynchronizable)
    }
```

This also means that the other `set` funcs for Float, Double and Bool will also recurse since they hit the Int version.

**Comment by lmarceau** [comment]
Hello, this code is in another repository [here](https://github.com/mozilla/rust-components-swift) and maintained by another team. You can open an issue on their end or open a PR there if you wish to.

**Reply by gutley** [reply]
Thanks - [Raised](https://github.com/mozilla/rust-components-swift/issues/99) over there.
